src/instruction.rs

```diff
@@ -183,7 +183,7 @@ mod test {
         let merchant_keypair = Keypair::new();
 
         let (mut banks_client, payer, recent_blockhash) = ProgramTest::new(
-            "solana_payment_processor",
+            "sol_payment_processor",
```

Comment:
If the name here doesn't match the name of the file in `./target/deploy/.` then the program test framework will run your program natively instead of in the BPF VM. With this change, you won't run into the buggy invoke behavior